openmc_plasma_source/tokamak_source.py

```diff
@@ -46,50 +54,90 @@ class TokamakSource:
-        minor_radius,
-        elongation,
-        triangularity,
-        mode,
-        ion_density_centre,
-        ion_density_peaking_factor,
-        ion_density_pedestal,
-        ion_density_separatrix,
-        ion_temperature_centre,
+        major_radius: float,
+        minor_radius: float,
+        elongation: float,
+        triangularity: float,
+        mode: str,
+        ion_density_centre: float,
+        ion_density_peaking_factor: float,
+        ion_density_pedestal: float,
+        ion_density_separatrix: float,
+        ion_temperature_centre: float,
         ion_temperature_peaking_factor,
         ion_temperature_beta,
-        ion_temperature_pedestal,
-        ion_temperature_separatrix,
-        pedestal_radius,
-        shafranov_factor,
+        ion_temperature_pedestal: float,
+        ion_temperature_separatrix: float,
+        pedestal_radius: float,
+        shafranov_factor: float,
         angles=(0, 2 * np.pi),
-        sample_size=1000,
+        sample_size: int = 1000,
     ) -> None:
+        # Assign attributes
         self.major_radius = major_radius
         self.minor_radius = minor_radius
         self.elongation = elongation
         self.triangularity = triangularity
         self.ion_density_centre = ion_density_centre
-        self.mode = mode
         self.ion_density_peaking_factor = ion_density_peaking_factor
+        self.mode = mode
         self.pedestal_radius = pedestal_radius
         self.ion_density_pedestal = ion_density_pedestal
         self.ion_density_separatrix = ion_density_separatrix
-
         self.ion_temperature_centre = ion_temperature_centre
         self.ion_temperature_peaking_factor = ion_temperature_peaking_factor
         self.ion_temperature_pedestal = ion_temperature_pedestal
         self.ion_temperature_separatrix = ion_temperature_separatrix
         self.ion_temperature_beta = ion_temperature_beta
-
         self.shafranov_factor = shafranov_factor
-
+        self.angles = angles
         self.sample_size = sample_size
 
-        self.angles = angles
+        # Perform sanity checks for inputs not caught by properties
+        if self.minor_radius >= self.major_radius:
+            raise ValueError("Minor radius must be smaller than major radius")
+
+        if self.pedestal_radius >= self.minor_radius:
+            raise ValueError("Pedestal radius must be smaller than minor radius")
+
+        if abs(self.shafranov_factor) >= 0.5 * minor_radius:
+            raise ValueError("Shafranov factor must be smaller than 0.5*minor radius")
+
+        if len(self.angles) != 2:
+            raise ValueError(
+                "TokamakSource.angles must be set to a list/tuple of length 2."
+            )
+        try:
+            float(self.angles[0])
+            float(self.angles[1])
+        except ValueError:
+            raise ValueError(
+                "TokamakSource.angles: minimum and maximum angles must be "
+                "convertable to float."
```

Comment:
```suggestion
                "convertible to float."
```

Reply:
Well caught! I think I may have been spelling that incorrectly for years.